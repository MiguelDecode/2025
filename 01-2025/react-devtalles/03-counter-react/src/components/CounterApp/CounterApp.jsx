import PropTypes from "prop-types";
import { useState } from "react";

export const CounterApp = ({ initialValue }) => {
  const [count, setCount] = useState(initialValue);

  const handleAdd = () => {
    setCount(count + 1);
  };

  const handleSubstract = () => {
    setCount(count - 1);
  };

  const handleReset = () => {
    setCount(initialValue);
  };

  return (
    <>
      <h4>CounterApp</h4>
      <h5>{count}</h5>
      <button onClick={handleAdd}>+1</button>
      <button onClick={handleSubstract}>-1</button>
      <button aria-label="btn-reset"  onClick={handleReset}>
        Reset
      </button>
    </>
  );
};

CounterApp.propTypes = {
  initialValue: PropTypes.number,
};
