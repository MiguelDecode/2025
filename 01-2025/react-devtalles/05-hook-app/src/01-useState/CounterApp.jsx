import { useState } from "react";

export const CounterApp = () => {
  const [counter, setCounter] = useState({
    counter1: 10,
    counter2: 20,
    counter3: 30,
  });

  const { counter1, counter2, counter3 } = counter;

  const handleAdd = () => {
    setCounter({
      counter1: counter1 + 1,
      counter2: counter2 + 2,
      counter3: counter3 + 3,
    });
  };

  return (
    <>
      <h2>Counter x 1: {counter1}</h2>

      <h2>Counter x 2: {counter2}</h2>

      <h2>Counter x 3: {counter3}</h2>

      <hr />

      <button className="btn" onClick={handleAdd}>
        +1
      </button>
    </>
  );
};
