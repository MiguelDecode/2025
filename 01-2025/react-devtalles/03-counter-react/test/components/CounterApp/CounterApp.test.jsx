import { render, screen, fireEvent } from "@testing-library/react";
import { CounterApp } from "../../../src/components/CounterApp/CounterApp";

describe("Tests on component <CounterApp/>", () => {
  const initialValue = 100;

  test("should do match with the snapshot", () => {
    const { container } = render(<CounterApp initialValue={initialValue} />);

    expect(container).toMatchSnapshot();
  });

  test("should show a initial value", () => {
    render(<CounterApp initialValue={initialValue} />);

    expect(screen.getByText(100)).toBeTruthy();
    expect(screen.getByRole("heading", { level: 5 }).innerHTML).toContain(
      String(initialValue)
    );
  });

  test("should increment with the button +1", () => {
    render(<CounterApp initialValue={initialValue} />);

    fireEvent.click(screen.getByText("+1"));

    expect(screen.getByRole("heading", { level: 5 }).innerHTML).toContain(
      String(initialValue + 1)
    );
  });

  test("should substract with the button -1", () => {
    render(<CounterApp initialValue={initialValue} />);

    fireEvent.click(screen.getByText("-1"));

    expect(screen.getByRole("heading", { level: 5 }).innerHTML).toContain(
      String(initialValue - 1)
    );
  });

  test("should reset with the button reset", () => {
    render(<CounterApp initialValue={initialValue} />);

    fireEvent.click(screen.getByText("+1"));
    fireEvent.click(screen.getByText("+1"));
    fireEvent.click(screen.getByText("+1"));
    fireEvent.click(screen.getByText("+1"));

    // fireEvent.click(screen.getByText("Reset"));
    fireEvent.click(screen.getByLabelText("btn-reset"));

    expect(screen.getByRole("heading", { level: 5 }).innerHTML).toContain(
      String(initialValue)
    );
  });
});
