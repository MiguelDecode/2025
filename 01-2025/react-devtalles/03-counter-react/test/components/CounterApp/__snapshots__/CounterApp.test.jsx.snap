// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on component <CounterApp/> should do match with the snapshot 1`] = `
<div>
  <h4>
    CounterApp
  </h4>
  <h5>
    100
  </h5>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button
    aria-label="btn-reset"
  >
    Reset
  </button>
</div>
`;
